File: leetcode/277.find-the-celebrity.cpp
// Forward declaration of the knows API.
bool knows(int a, int b);

class Solution {
public:
    int findCelebrity(int n) {
        
        int candidate = 0;
        for(auto i = 1; i < n; ++i)
        {
            candidate = knows(candidate, i) ? i: candidate;
        }

        for (auto i = 0; i < n; ++i)
        {
            if (candidate == i)
            {
                continue;
            }

            if(knows(candidate, i))
            {
                return -1;
            }

            if (!knows(i, candidate))
            {
                return -1;
            }
        }

        return candidate;
    }
};

File: leetcode/277.find-the-celebrity.py
# The knows API is already defined for you.
# @param a, person a
# @param b, person b
# @return a boolean, whether a knows b
# def knows(a, b):

class Solution(object):
    def findCelebrity(self, n):
        """
        :type n: int
        :rtype: int
        """
        candidate = 0
        for i in range(1, n):
            if knows(candidate, i):
                candidate = i

        for i in range(n):
            if i == candidate:
                continue

            if knows(candidate, i):
                return -1

            if not knows(i, candidate):
                return -1

        return candidate